XpyriMentor/suggestors/sequential_strategizer.py

```diff
@@ -0,0 +1,109 @@
+import math
+import warnings
+from typing import Iterable
+
+import numpy as np
+
+from .default_suggestor import DefaultSuggestor
+from .lhs_suggestor import LHSSuggestor
+from .po_suggestor import POSuggestor
+from .suggestor import IncompatibleNumberAsked, Suggestor
+
+
+class SequentialStrategizer():
+    """
+    Stratgizer that uses a sequence of suggestors, each with a budget of suggestions to
+    make.
+
+    It uses the suggestors in order, skipping suggestors with a budget of suggestions
+    that have already been made.
+    """
+    def __init__(self, suggestors: list[tuple[int, Suggestor]]):
+        """
+        Initialize the strategizer with a list of suggestors and their budgets.
+
+        Parameters
+        ----------
+        * suggestors [`list[tuple[int, Suggestor]]`]:
+            A list of tuples where the first element is the number of suggestions the
+            suggestor can make and the second element is the suggestor. A negative number
+            of suggestions is interpreted as infinity, aand can only be used as the last
+            budget.
+
+            If the first suggestor is a DefaultSuggestor, it will be replaced with a
+            LHSSuggestor with the same budget. If any other suggestor is a
+            DefaultSuggestor, it will be replaced with a POSuggestor.
+        """
+        for n, (budget, suggestor) in enumerate(suggestors):
+            if isinstance(suggestor, DefaultSuggestor):
+                if n == 0:
+                    suggestors[n] = (
+                        budget,
+                        LHSSuggestor(
+                            space=suggestor.space, rng=suggestor.rng, n_points=budget
+                        )
+                    )
+                else:
+                    suggestors[n] = (
+                        budget,
+                        POSuggestor(
+                            space=suggestor.space,
+                            rng=suggestor.rng,
+                            n_objectives=suggestor.n_objectives,
+                        ))
+            if budget <= 0:
+                # Interpret negative budgets as infinity
+                suggestors[n] = (float("inf"), suggestors[n][1])
+            if hasattr(suggestors[n][1], "n_points"):
+                if suggestors[n][1].n_points != suggestors[n][0]:
+                    warnings.warn(
+                        f"Budget of {suggestors[n][0]} points does not match number of "
+                        "points for suggestor of type "
+                        f"{suggestors[n][1].__class__.__name__}."
+                    )
+            if math.isinf(suggestors[n][0]):
+                if n < len(suggestors) - 1:
+                    raise ValueError("Infinite budget must be the last budget")
```

Comment:
Change Error msg: Only the last suggestor can have infinite budget

Reply:
Done, much better phrasing.